lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -354,87 +386,117 @@ The log data.
 
 =back
 
-Return: processed data
-
+Returns processed data
 
 =cut
 
 sub _filter_stack {
     my ($self, $data) = @_;
-    return $data if(numeric_level($data->{severity}) <= numeric_level('warn'));
+
+    return $data if (numeric_level($data->{severity}) <= numeric_level('warn'));
+
     # now severity > warn
     return $data if $self->{log_level} >= numeric_level('debug');
+
     delete $data->{stack};
     return $data;
 }
 
 =head2 _collapse_future_stack
 
-The future frames are too much and too tedious. This method will keep only one frame if there are many continuously future frames.
-Parameter: log data
-Return: log data
+The future L<FUTURE> frames are too much and too tedious. This method will keep only one
+frame if there are many continuously future frames.
+
+Takes the following arguments as named parameters
+
+=over 4
+
+=item * C<$data> Log stack data
+
+=back
+
+Returns log data
 
 =cut
 
-sub _collapse_future_stack{
+sub _collapse_future_stack {
     my ($self, $data) = @_;
     my $stack = $data->{stack};
     my @new_stack;
     my $previous_is_future;
-    for my  $frame ($stack->@*){
-        if($frame->{package} eq 'Future'){
-            next if($previous_is_future);
+
+    for my $frame ($stack->@*) {
+        if ($frame->{package} eq 'Future') {
+            next if ($previous_is_future);
             push @new_stack, $frame;
             $previous_is_future = 1;
-        }
-        else{
+        } else {
             push @new_stack, $frame;
             $previous_is_future = 0;
         }
     }
     $data->{stack} = \@new_stack;
+
     return $data;
 }
 
+=head2 _fh_is_tty
+
+Check the filehandle opened to tty
+Returns boolean
+
+=cut
+
 sub _fh_is_tty {
     my $fh = shift;
-   return -t $fh;
+
+    return -t $fh;
```

Comment:
Why do we need this function at all? This is a standard Perl feature, can we not use it directly?

Reply:
We use it  because we can mock this function be easily in the test file